File: LES_EXAMENS/examen_de_rattrapage_2023_2024/exercice_F.py
T=[None for i in range(99)]
while True:
    n=int(input("saisir un nombre entier n<100: "))
    if n<100 :
        break
for i in range(0,n):
    T[i]=float(input(f"saisir le reel numero {i+1}: "))
print("{",end="")
for i in range(0,n):
    print(f" {T[i]} ",end="")
print("}")
min=T[0]
max=T[0]
for i in range(1,n):
    if T[i]>max:
        max=T[i]
    if T[i]<min:
        min=T[i]
print(f"la valeur min de le tableau est {min}")
print(f"la valeur max de le tableau est {max}")
divis=max-min
for i in range(0,n):
    T[i]=T[i]/divis
print("{",end="")
for i in range(0,n):
    print(f" {T[i]:.3f} ",end="")
print("}")

File: LES_EXAMENS/examen_de_rattrapage_2023_2024/exercice_F.c
#include <stdio.h>
float max,min,devis;
int n;
float T[99];
int main(){
    //saisir et affich le tableau T
    do{
        printf("saisir un entier n<100: ");
        scanf("%d",&n);
    }while(n<0 || n>=100);
    for(int i=0;i<n;i++){
        printf("saisir la valeur %d du tableau: ",i+1);
        scanf("%f",&T[i]);
    }
    printf("{");
    for(int i=0;i<n;i++){
        printf(" %f ",T[i]);
    }
    printf("}\n");
    //chaircher et afficher min et max
    min=T[0];
    max=T[0];
    for(int i=0;i<n;i++){
        if(T[i]>max){
            max=T[i];
        }
        if(T[i]<min){
            min=T[i];
        }
    }
    printf("min=%f\n",min);
    printf("max=%f\n",max);
    //Normalise et afficher toute les valeurs du tableau 0-1
    devis=max-min;
    for(int i=0;i<n;i++){
        T[i]=T[i]/devis;
    }
    printf("{");
    for(int i=0;i<n;i++){
        printf(" %f ",T[i]);
    }
    printf("}\n");
    return 0;
}